### ISSUE TITLE
Incorrect packaging Grading terminated.

### ISSUE THREAD

**Issue by throwawayAcc123** [nus-cs2030/2122-s1#216]
Hi guys, I'm facing this problem when submitting to CodeCrunch. For every file except for the MainN files, I included a line of package cs2030.simulator, and I included import cs2030.simulator.Simulator for all the MainN files .

But still, this problem arises. Appreciate any help, thank you!

**Comment by Eclipse-Dominator** [comment]
Make sure all your project files has `package cs2030s.simulator` for all your other project files

**Comment by throwawayAcc123** [comment]
Yup,other than MainN files, I have that line for all my project files

**Comment by BrendanCheong** [comment]
personally my project files should look something like this

📦Project
┣ 📂cs2030
┃ ┗ 📂simulator
┃ ┃ ┣ 📜Event.class
┃ ┃ ┣ 📜RandomGenerator.class
┃ ┃ ┗ 📜Simulator.class
┣ 📜Event.java
┣ 📜Main.class
┣ 📜Main.java
┗ 📜Simulator.java

first make sure all your java files are under /Project and all files (except Main.java) has ```package cs2030.simulator;``` as its first line.
Make sure your Main.java has ```import cs2030.simulator.Simulator;```
Now run ```javac -d . *.java``` in your /Project folder
You should see cs2030/simulator folder appearing, here you can find all your ```.class``` files for java files that had that ```package cs2030.simulator;``` line
Now, place your ```RandomGenerator.class``` file in that cs2030/simulator folder.

Hopefully this helps

**Comment by jehousoh** [comment]
just to check are your package `cs2030.simulator;` line at the top of each class?

**Comment by dragane359** [comment]
Try adding `import cs2030.simulator.*;` to MainN.java as well maybe

**Comment by yuyanjulia** [comment]
Maybe can refer to these two #156 #176 and see how they have solved the issue

**Comment by KeithPJX** [comment]
Remember to compile using `javac -d . *.java`

**Comment by sheshenk** [comment]
I had this issue when I included `Main.java` (the level 0 file) in my upload to CodeCrunch. Maybe you could try excluding this file and try again?

**Comment by JadeHZY** [comment]
Here is the advice from my TA, and I found it very helpful. Hope it'll help those still facing the issue:

**1. The MainX files should not be packaged
2. Only the Simulator class that MainX uses should be public and imported into MainX
3. The RandomGenerator class file should be placed inside the cs2030/simulator folder together with all of the other class files
4. The RandomGenerator can only be used with classes in the same package (i.e. it cannot be used inside MainX)**

**Comment by Lokki-ki-ki** [comment]
> I had this issue when I included `Main.java` (the level 0 file) in my upload to CodeCrunch. Maybe you could try excluding this file and try again?

Thanks!! It solved my issue as I also included Main.java
